src/app.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import Button from './button';
+import ButtonList from './buttonList';
+import { MemoizedTitle } from './title';
+
+const numbers = (numberLength) => new Array(numberLength).fill().map((_, index) => index + 1);
+
+function App() {
+  const [count, setCount] = useState(0);
+  const [maxButtonNumber, setMaxButtonNumber] = useState(5);
+  const buttonNumbers = useMemo(() => numbers(maxButtonNumber), [maxButtonNumber]);
+
+  const handleClick = useCallback((number) => {
+    setCount((previousCount) => previousCount + number);
+  });
+
+  const handleAddNumberButtonClick = useCallback(() => {
+    setMaxButtonNumber((previousNumber) => previousNumber + 1);
+  });
+
+  return (
+    <>
+      <h1>CodeSoom - week2 assignment - 1</h1>
+      <MemoizedTitle text="Counter" />
+      <Button onClick={() => handleClick(1)}>{`Click me!(${count})`}</Button>
+      <ButtonList numbers={buttonNumbers} onClick={handleClick} />
+      <br />
+      <Button onClick={handleAddNumberButtonClick}>Add number to button list!</Button>
```

Comment:
title 과 button 은 의미있는 태그로 나누면 좋을거같아요! 예를들면 `<p>` 태그를 사용할수 있을거같아요!
```suggestion
      <p>
        <Button onClick={() => handleClick(1)}>{`Click me!(${count})`}</Button>
      </p>
      <p>
        <ButtonList numbers={buttonNumbers} onClick={handleClick} />
      </p>
```

Reply:
두 버튼은 무엇 때문에 의미있는 태그로 나눠야 하는건가요??
 혹시 레이아웃 때문이면 <p>가 아닌 <div>를 사용해야 되는 상황이 아닌가요?

Reply:
시멘틱 태그를 사용하는건 항상 어려운거 같아요...! 
개인적으로는 div 는 경계를 나눈다고 생각하고, p 태그는 단락 이라고 생각을해요!
`div 로 나눈다 -> 나눠지는 영역이 서로 관련이 없다 ` 라고 생각을 합니다! (실무에서는 여기까지 신경을 쓰고있지 못하고있어요 🤧)

Reply:
넵, 참고하겠습니다!